content/en/blog/_posts/2022-04-19-csi-nodeexpandsecret.md

```diff
@@ -0,0 +1,183 @@
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  annotations:
+    pv.kubernetes.io/provisioned-by: hostpath.csi.k8s.io
+  creationTimestamp: "2022-03-23T15:14:07Z"
+  finalizers:
+  - kubernetes.io/pv-protection
+  name: pvc-95eb531a-d675-49f6-940b-9bc3fde83eb0
+  resourceVersion: "420263"
+  uid: 6fa824d7-8a06-4e0c-b722-d3f897dcbd65
+spec:
+  accessModes:
+  - ReadWriteOnce
+  capacity:
+    storage: 6Gi
+  claimRef:
+    apiVersion: v1
+    kind: PersistentVolumeClaim
+    name: csi-pvc
+    namespace: default
+    resourceVersion: "419862"
+    uid: 95eb531a-d675-49f6-940b-9bc3fde83eb0
+  csi:
+    driver: hostpath.csi.k8s.io
+    nodeExpandSecretRef:
+      name: test-secret
+      namespace: default
+    volumeAttributes:
+      storage.kubernetes.io/csiProvisionerIdentity: 1648042783218-8081-hostpath.csi.k8s.io
+    volumeHandle: e21c7809-aabb-11ec-917a-2e2e254eb4cf
+  nodeAffinity:
+    required:
+      nodeSelectorTerms:
+      - matchExpressions:
+        - key: topology.hostpath.csi/node
+          operator: In
+          values:
+          - rook-node01
+  persistentVolumeReclaimPolicy: Delete
+  storageClassName: csi-hostpath-sc
+  volumeMode: Filesystem
+status:
+  phase: Bound
+```
+
+Expand the volume online, the secret will be pass to the csi driver, like this,
+the logs in host path csi driver:
+
+```yaml
+I0330 03:29:51.966241       1 server.go:101] GRPC call: /csi.v1.Node/NodeExpandVolume
+I0330 03:29:51.966261       1 server.go:105] GRPC request: {"capacity_range":{"required_bytes":7516192768},"secrets":"***stripped***","staging_target_path":"/var/lib/kubelet/plugins/kubernetes.io/csi/hostpath.csi.k8s.io/f7c62e6e08ce21e9b2a95c841df315ed4c25a15e91d8fcaf20e1c2305e5300ab/globalmount","volume_capability":{"AccessType":{"Mount":{}},"access_mode":{"mode":7}},"volume_id":"e21c7809-aabb-11ec-917a-2e2e254eb4cf","volume_path":"/var/lib/kubelet/pods/bcb1b2c4-5793-425c-acf1-47163a81b4d7/volumes/kubernetes.io~csi/pvc-95eb531a-d675-49f6-940b-9bc3fde83eb0/mount"}
+I0330 03:29:51.966360       1 nodeserver.go:459] req:volume_id:"e21c7809-aabb-11ec-917a-2e2e254eb4cf" volume_path:"/var/lib/kubelet/pods/bcb1b2c4-5793-425c-acf1-47163a81b4d7/volumes/kubernetes.io~csi/pvc-95eb531a-d675-49f6-940b-9bc3fde83eb0/mount" capacity_range:<required_bytes:7516192768 > staging_target_path:"/var/lib/kubelet/plugins/kubernetes.io/csi/hostpath.csi.k8s.io/f7c62e6e08ce21e9b2a95c841df315ed4c25a15e91d8fcaf20e1c2305e5300ab/globalmount" volume_capability:<mount:<> access_mode:<mode:SINGLE_NODE_MULTI_WRITER > > secrets:<key:"XXXXXX" value:"XXXXX" > secrets:<key:"XXXXX" value:"XXXXXX" >
+```
+
+## The future
+
+As this feature is still in alpha, we expect to update or get feedback from CSI driver
+authors with more tests and implementation. The community plans to eventually
+promote the feature to Beta in upcoming releases.
+
+## How can I learn more?
```

Comment:
```suggestion
## Get involved or learn more
```
“I” doesnt make sense. You'd be switching from “we” representing the authors / contributors to “I” representing the reader, which jars.